server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -23,46 +23,71 @@
 
 import io.crate.common.collections.MapComparator;
 import io.crate.data.Input;
-import io.crate.expression.symbol.Function;
-import io.crate.expression.symbol.Symbol;
-import io.crate.metadata.BaseFunctionResolver;
 import io.crate.metadata.FunctionIdent;
-import io.crate.metadata.FunctionImplementation;
 import io.crate.metadata.FunctionInfo;
 import io.crate.metadata.TransactionContext;
-import io.crate.metadata.functions.params.FuncParams;
-import io.crate.metadata.functions.params.Param;
-import io.crate.types.DataType;
-import io.crate.types.DataTypes;
-import io.crate.types.ObjectType;
-
-import java.util.Arrays;
-import java.util.List;
+import io.crate.metadata.functions.Signature;
+
+import javax.annotation.Nullable;
 import java.util.Map;
 
-import static io.crate.expression.operator.CmpOperator.CmpResolver.createInfo;
+import static io.crate.metadata.functions.TypeVariableConstraint.typeVariable;
+import static io.crate.types.TypeSignature.parseTypeSignature;
 
 public final class EqOperator extends Operator<Object> {
 
     public static final String NAME = "op_=";
 
-    private final FunctionInfo info;
+    public static final Signature SIGNATURE = Signature.scalar(
+        NAME,
+        parseTypeSignature("E"),
+        parseTypeSignature("E"),
+        Operator.RETURN_TYPE.getTypeSignature()
+    ).withTypeVariableConstraints(typeVariable("E"));
+
+    public static final Signature SIGNATURE_OBJECTS = Signature.scalar(
+        NAME,
+        parseTypeSignature("object(K, V)"),
+        parseTypeSignature("object(K, V)"),
```

Comment:
just a question: why is matched with `object(k, v)` shouldn't we match it with just `object` such as the map can contain different `v` of different types?

Reply:
good point, will check.

Reply:
Figured out that we don't need the special `ObjectEqOperator` at all as our `ObjectType.equals` is checking also inner types and our signature binder allows coercion between un-typed objects and typed once.